hikari/channels.py

```diff
@@ -95,6 +97,95 @@ def __str__(self) -> str:
         return self.name
 
 
+@attr_extensions.with_copy
+@attr.s(eq=True, hash=True, init=True, kw_only=True, slots=True, weakref_slot=False)
+class ChannelFollow:
+    """Represents a channel follow."""
+
+    app: traits.RESTAware = attr.ib(repr=False, eq=False, hash=False, metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    """The client application that models may use for procedures."""
+
+    channel_id: snowflakes.Snowflake = attr.ib(eq=True, hash=True, repr=True)
+    """The ID of the news channel that's being followed."""
+
+    webhook_id: snowflakes.Snowflake = attr.ib(eq=True, hash=True, repr=True)
+    """The ID of the webhook for this follow."""
+
+    async def fetch_channel(self) -> GuildChannel:
+        """Fetch the object of the guild channel being followed.
+
+        Returns
+        -------
+        hikari.channels.PartialChannel
+            The channel being followed. This will be a _derivative_ of
+            `hikari.channels.PartialChannel`, depending on the type of
+            channel you request for.
+
+            This means that you may get one of
+            `hikari.channels.PrivateTextChannel`,
+            `hikari.channels.GroupPrivateTextChannel`,
+            `hikari.channels.GuildTextChannel`,
+            `hikari.channels.GuildVoiceChannel`,
```

Comment:
~~How does following a voice channel even work?~~

~~Being able to follow a DM channel seems worrying...~~

nvm, assume this was a mistake from the type hints?

Reply:
@davfsa you'll need to remember to rebase onto this and update this naming if you are changing `PrivateChannel` objects to `DMChannel` names.

Reply:
Im holding back on doing it till this gets merged as im working on other stuff